src/components/TimeseriesChart/TimeseriesChart.tsx

```diff
@@ -107,6 +127,19 @@ export class TimeseriesChart extends React.Component<
     }
   };
 
+  onMouseMove = (data: any) => {
+    const { points = [] } = data;
+    const legendValues: LegendValues = {};
+    points.forEach((point: LegendValue) => {
+      legendValues[point.id] = point;
+    });
+    this.setState({ legendValues });
+
+    if (typeof this.props.onMouseMove === 'function') {
```

Comment:
Use destruction assigment for `onMouseMove` prop

Reply:
`typeof this.props.onMouseMove === 'function'` can be replaced with just `if (this.props.onMouseMove` if you have proper types

Reply:
@mrtraser - "Use destruction assignment"?

@polomani - I was thinking of the component being used by a JS client which won't catch if a non-function is passed to `onMouseMove` prop.

Reply:
yes, but we don't spend our time on that
you will never be able to do all type checks. it's endless list
they should use TS

Reply:
> Use destruction assignment

```typescript
const { onMouseMove } = this.props
```

Reply:
Used destruction assignment.

It is either:
```js
const { onMouseMove } = this.props;
if (typeof onMouseMove === 'function') {
      onMouseMove(data);
}
```
Or
```js
const { onMouseMove } = this.props;
if (onMouseMove) {
      onMouseMove(data);
}
```

Since we have a check anyway, going with the first version.